### PR TITLE
Add ContactDataSource

### PR DESCRIPTION
**Pull request by wmontwe**
Extract `ContentResolver` use of `Contacts` into `ContactDataSource` and introduce `Contact` entity. Requests for name, uri and photoUri are joined into one call.

Depends on #6727

### CODE REVIEW

File: core/android/common/src/main/kotlin/app/k9mail/core/android/common/database/CursorExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package app.k9mail.core.android.common.database
```
Comment (cketti) [inline core/android/common/src/main/kotlin/app/k9mail/core/android/common/database/CursorExtensions.kt:1]: We already have a `CursorExtensions.kt` file. Maybe these should be combined?

File: app/core/src/main/java/com/fsck/k9/helper/Contacts.kt [hunk 2]
```diff
@@ -80,17 +54,14 @@ open class Contacts(
...
-            }
-            cursor.close()
+
+        val contact = contactDataSource.getContactFor(emailAddress)
+        return if (contact != null) {
+            nameCache[emailAddress] = contact.name
+            contact.name
+        } else {
+            null
         }
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply app/core/src/main/java/com/fsck/k9/helper/Contacts.kt:58]: Good catch, it should cache empty results too. Will fix this.

File: core/android/common/src/main/kotlin/app/k9mail/core/android/common/database/CursorExtensions.kt (hunk 1 above)
Comment (wmontwe) [reply core/android/common/src/main/kotlin/app/k9mail/core/android/common/database/CursorExtensions.kt:1]: I'll have a look

Comment (wmontwe) [reply core/android/common/src/main/kotlin/app/k9mail/core/android/common/database/CursorExtensions.kt:1]: I joined them

**Review by cketti — APPROVED** [review]

**Pull request merged**